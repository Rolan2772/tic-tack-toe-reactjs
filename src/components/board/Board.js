import React from "react";
import Square from "../square/Square";
import Grid from "@material-ui/core/Grid";
import {withStyles} from '@material-ui/core/styles';

const styles = () => ({
    boardMargin: {
        margin: '10px'
    }
});

export const DynamicRow = (props) => {
    const {rowIndex, rowData, onClick} = props;
    const size = rowData.length;
    let cells = [];

    for (let index = 0; index < size; index++) {
        const cellIndex = rowIndex * size + index;
        const value = rowData[index];
        cells.push(<Square key={cellIndex} value={value} onClick={() => onClick(cellIndex)}/>);
    }
    return (
        <Grid container justify="center">
            {cells}
        </Grid>
    )
};

export const DynamicBoard = (props) => {
    const {boardData, onClick} = props;
    const board = [];
    const boardSize = Math.sqrt(boardData.length);

    for (let rowIndex = 0; rowIndex < boardSize; rowIndex++) {
        const start = rowIndex * boardSize;
        const end = start + boardSize;
        const rowData = boardData.slice(start, end);
        board.push(<DynamicRow key={rowIndex} rowIndex={rowIndex} rowData={rowData} onClick={onClick}/>);
    }
    return board;
};

export const Board = (props) => {
    const {squares, onClick} = props;

    return (
        <Grid className={props.classes.boardMargin}>
            {<DynamicBoard boardData={squares} onClick={onClick}/>}
        </Grid>
    )
};

export default Board;