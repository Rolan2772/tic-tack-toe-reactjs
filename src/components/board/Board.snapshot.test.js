import React from 'react';
import Board from './Board';
import renderer from 'react-test-renderer';

describe('<Board />', () => {

    it('should match snapshot', () => {
        const squares = [];
        const component = renderer.create(
            <Board squares={squares} onclick={() => ''}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

