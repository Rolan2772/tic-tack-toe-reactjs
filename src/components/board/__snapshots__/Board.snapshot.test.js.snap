// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> should match snapshot 1`] = `
<div
  className="Board-boardMargin-1"
/>
`;
